### ISSUE TITLE
Redundant conditional in ecdsa_impl.h

### ISSUE THREAD

**Issue by bytting** [bitcoin-core/secp256k1#467]
The following code block can be seen in src/secp256k1/src/ecdsa_impl.h starting at line 83

```
while (lenleft > 0) {
    if ((ret >> ((sizeof(size_t) - 1) * 8)) != 0) {
    }
    ret = (ret << 8) | **sigp;
    if (ret + lenleft > (size_t)(sigend - *sigp)) {
        /* Result exceeds the length of the passed array. */
        return -1;
    }
    (*sigp)++;
    lenleft--;
}
```

The conditional _if_ statement at line 84 doesn't appear to have a purpose other than spending cycles

**Comment by sipa** [comment]
Eh, agree...

**Comment by corebob** [comment]
Yeah, probably. I was wondering if the line right after it should actually have been inside, but I can't really tell. Seems unlikely though.

Is it worth making a PR?

**Comment by apoelstra** [comment]
Looks like this should've been closed when #468 was merged.
